beets/importer.py

```diff
@@ -1225,7 +1225,33 @@ def handlers(cls):
             cls._handlers.append((is_zipfile, ZipFile))
             import tarfile
 
-            cls._handlers.append((tarfile.is_tarfile, tarfile.open))
+            class TarInfo(tarfile.TarInfo):
```

Comment:
I am not a python dev. Not sure this is correct, but is working on my install.

Reply:
The idea right correct; however I think the definition of `TarInfo` and `TarWrapper` should be moved out of `set_choice` directly to the top-level of this file.